src/scheduler_utils.cc

```diff
@@ -237,6 +237,32 @@ PriorityQueue::PolicyQueue::ApplyPolicy(
   return ((idx - queue_.size()) < delayed_queue_.size());
 }
 
+size_t
+PriorityQueue::PolicyQueue::RejectTimeoutRequests()
+{
+  if (timeout_action_ != inference::ModelQueuePolicy::REJECT) {
+    return false;
+  }
+
+  size_t rejected_count = 0;
+  uint64_t now_nanoseconds =
+      std::chrono::duration_cast<std::chrono::nanoseconds>(
+          std::chrono::steady_clock::now().time_since_epoch())
+          .count();
+  for (size_t idx = 0; idx < queue_.size();) {
```

Comment:
maybe while loop with some end condition flag instead of a for loop? I was concerned/confused in the beginning regarding the running index, but then noticed you control it yourself

Reply:
Sure, switched to while loop.